core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -5843,60 +5843,60 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                         PUT_V128_TO_ADDR(frame_lp + addr_ret, LOAD_V128(maddr));
                         break;
                     }
-#define SIMD_LOAD_OP(simde_func, element_size, num_elements) \
-    do {                                                     \
-        uint32 offset, addr;                                 \
-        offset = read_uint32(frame_ip);                      \
-        addr = GET_OPERAND(uint32, I32, 0);                  \
-        frame_ip += 2;                                       \
-        addr_ret = GET_OFFSET();                             \
-        CHECK_MEMORY_OVERFLOW(16);                           \
-                                                             \
-        simde_v128_t simde_result = simde_func(maddr);       \
-                                                             \
-        V128 result;                                         \
-        SIMDE_V128_TO_SIMD_V128(simde_result, result);       \
-        PUT_V128_TO_ADDR(frame_lp + addr_ret, result);       \
-                                                             \
+#define SIMD_LOAD_OP(simde_func, width)                \
+    do {                                               \
+        uint32 offset, addr;                           \
+        offset = read_uint32(frame_ip);                \
+        addr = GET_OPERAND(uint32, I32, 0);            \
+        frame_ip += 2;                                 \
+        addr_ret = GET_OFFSET();                       \
+        CHECK_MEMORY_OVERFLOW(width / 8);              \
```

Comment:
is that correct? e.g. for `SIMD_v128_load8x8_s` we'll only check if single byte can fit into the address? Shouldn't we check for 16bytes-slot?